src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -0,0 +1,289 @@
+    PMACC_ALIGN(cachedB, DataBox<SharedBox<ValueType_B, typename BlockArea::FullSuperCellSize,0> >);
+
+    PMACC_ALIGN(curF_1, SynchrotronFunctions::SyncFuncCursor);
+    PMACC_ALIGN(curF_2, SynchrotronFunctions::SyncFuncCursor);
+
+    /* random number generator for Monte Carlo */
+    typedef ionization::RandomNrForMonteCarlo<T_ElectronSpecies> RandomGen;
+    /* \todo fix: cannot PMACC_ALIGN() because it seems to be too large */
+    RandomGen randomGen;
+
+    float3_X photon_mom;
+
+public:
+    /* host constructor initializing member : random number generator */
+    PhotonCreator(
+        const SynchrotronFunctions::SyncFuncCursor& curF_1,
+        const SynchrotronFunctions::SyncFuncCursor& curF_2,
+        const uint32_t currentStep)
+            : curF_1(curF_1), curF_2(curF_2), randomGen(currentStep)
+    {
+        DataConnector &dc = Environment<>::get().DataConnector();
+        /* initialize pointers on host-side E-(B-)field databoxes */
+        FieldE* fieldE = &(dc.getData<FieldE > (FieldE::getName(), true));
+        FieldB* fieldB = &(dc.getData<FieldB > (FieldB::getName(), true));
+        /* initialize device-side E-(B-)field databoxes */
+        eBox = fieldE->getDeviceDataBox();
+        bBox = fieldB->getDeviceDataBox();
+    }
+
+    /** Initialization function on device
+     *
+     * \brief Cache EM-fields on device
+     *         and initialize possible prerequisites for ionization, like e.g. random number generator.
+     *
+     * This function will be called inline on the device which must happen BEFORE threads diverge
+     * during loop execution. The reason for this is the `__syncthreads()` call which is necessary after
+     * initializing the E-/B-field shared boxes in shared memory.
+     */
+    DINLINE void init(const DataSpace<simDim>& blockCell, const int& linearThreadIdx, const DataSpace<simDim>& localCellOffset)
+    {
+        /* caching of E and B fields */
+        cachedB = CachedBox::create < 0, ValueType_B > (BlockArea());
+        cachedE = CachedBox::create < 1, ValueType_E > (BlockArea());
+
+        /* instance of nvidia assignment operator */
+        nvidia::functors::Assign assign;
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldBBlock, bBox.shift(blockCell));
+        ThreadCollective<BlockArea> collective(linearThreadIdx);
+        collective(
+                  assign,
+                  cachedB,
+                  fieldBBlock
+                  );
+        /* copy fields from global to shared */
+        PMACC_AUTO(fieldEBlock, eBox.shift(blockCell));
+        collective(
+                  assign,
+                  cachedE,
+                  fieldEBlock
+                  );
+
+        /* wait for shared memory to be initialized */
+        __syncthreads();
+
+        /* initialize random number generator with the local cell index in the simulation*/
+        this->randomGen.init(localCellOffset);
```

Comment:
The random number generator used is this one: https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/particles/ionization/ionizationMethods.hpp
It uses the original RNG from picongpu with with the ionisation seed (and a few others) set up here: https://github.com/ComputationalRadiationPhysics/picongpu/pull/1354/files#diff-c51358fcd763da76cbd1cdd86809d342R110 and https://github.com/ComputationalRadiationPhysics/picongpu/blob/dev/src/picongpu/include/particles/ionization/ionizationMethods.hpp#L102

Short: The usage is correct. It would make sense to change the RNG in the class itself, but that change would be transparent and is for another PR